[harfbuzz] Build error on arm64-osx

Package: harfbuzz[core,coretext]:arm64-osx -> 3.2.0#1
**Host Environment**
- Host: arm64-osx
- Compiler: AppleClang 13.1.6.13160021
-    vcpkg-tool version: 2999-12-31-unknownhash-debug
    vcpkg-scripts version: a69b65229 2022-03-16 (9 hours ago)


**To Reproduce**
`vcpkg install `


**Failure logs**
```
-- Using cached harfbuzz-harfbuzz-3.2.0.tar.gz.
-- Cleaning sources at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/src/3.2.0-8c7a14e743.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/harfbuzz-harfbuzz-3.2.0.tar.gz
-- Applying patch 0001-circumvent-samefile-error.patch
-- Applying patch 0002-fix-uwp-build.patch
-- Using source at /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/src/3.2.0-8c7a14e743.clean
-- Getting CMake variables for arm64-osx-dbg
-- Getting CMake variables for arm64-osx-rel
-- Configuring arm64-osx-dbg
-- Configuring arm64-osx-dbg done
-- Configuring arm64-osx-rel
-- Configuring arm64-osx-rel done
-- Package arm64-osx-dbg
-- Package arm64-osx-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/downloads/tools/ninja-1.10.2-osx/ninja install -v
    Working Directory: /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/arm64-osx-rel
    Error code: 1
    See logs for more information:
      /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/package-arm64-osx-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:53 (vcpkg_execute_required_process)
  ports/harfbuzz/portfile.cmake:62 (vcpkg_install_meson)
  scripts/ports.cmake:145 (include)



```
<details><summary>/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/package-arm64-osx-rel-out.log</summary>

```
[1/65] /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/src/3.2.0-8c7a14e743.clean/src/gen-harfbuzzcc.py src/harfbuzz.cc /Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg/buildtrees/harfbuzz/src/3.2.0-8c7a14e743.clean/src ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-ankr-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-bsln-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-common.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-feat-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-just-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-kerx-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-morx-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-opbd-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout-trak-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout.cc ../src/3.2.0-8c7a14e743.clean/src/hb-aat-layout.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-ltag-table.hh ../src/3.2.0-8c7a14e743.clean/src/hb-aat-map.cc ../src/3.2.0-8c7a14e743.clean/src/hb-aat-map.hh ../src/3.2.0-8c7a14e743.clean/src/hb-algs.hh ../src/3.2.0-8c7a14e743.clean/src/hb-array.hh ../src/3.2.0-8c7a14e743.clean/src/hb-atomic.hh ../src/3.2.0-8c7a14e743.clean/src/hb-bimap.hh ../src/3.2.0-8c7a14e743.clean/src/hb-bit-page.hh ../src/3.2.0-8c7a14e743.clean/src/hb-blob.cc ../src/3.2.0-8c7a14e743.clean/src/hb-blob.hh ../src/3.2.0-8c7a14e743.clean/src/hb-buffer-serialize.cc ../src/3.2.0-8c7a14e743.clean/src/hb-buffer.cc ../src/3.2.0-8c7a14e743.clean/src/hb-buffer.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cache.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cff-interp-common.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cff-interp-cs-common.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cff-interp-dict-common.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cff1-interp-cs.hh ../src/3.2.0-8c7a14e743.clean/src/hb-cff2-interp-cs.hh ../src/3.2.0-8c7a14e743.clean/src/hb-common.cc ../src/3.2.0-8c7a14e743.clean/src/hb-config.hh ../src/3.2.0-8c7a14e743.clean/src/hb-debug.hh ../src/3.2.0-8c7a14e743.clean/src/hb-dispatch.hh ../src/3.2.0-8c7a14e743.clean/src/hb-draw.cc ../src/3.2.0-8c7a14e743.clean/src/hb-draw.hh ../src/3.2.0-8c7a14e743.clean/src/hb-face.cc ../src/3.2.0-8c7a14e743.clean/src/hb-face.hh ../src/3.2.0-8c7a14e743.clean/src/hb-fallback-shape.cc ../src/3.2.0-8c7a14e743.clean/src/hb-font.cc ../src/3.2.0-8c7a14e743.clean/src/hb-font.hh ../src/3.2.0-8c7a14e743.clean/src/hb-iter.hh ../src/3.2.0-8c7a14e743.clean/src/hb-kern.hh ../src/3.2.0-8c7a14e743.clean/src/hb-machinery.hh ../src/3.2.0-8c7a14e743.clean/src/hb-map.cc ../src/3.2.0-8c7a14e743.clean/src/hb-map.hh ../src/3.2.0-8c7a14e743.clean/src/hb-meta.hh ../src/3.2.0-8c7a14e743.clean/src/hb-ms-feature-ranges.cc ../src/3.2.0-8c7a14e743.clean/src/hb-ms-feature-ranges.hh ../src/3.2.0-8c7a14e743.clean/src/hb-mutex.hh ../src/3.2.0-8c7a14e743.clean/src/hb-null.hh ../src/3.2.0-8c7a14e743.clean/src/hb-number.cc ../src/3.2.0-8c7a14e743.clean/src/hb-number.hh ../src/3.2.0-8c7a14e743.clean/src/hb
...
Skipped 48 lines
...

[50/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include/libpng16 -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz.a.p/hb-style.cc.o -MF src/libharfbuzz.a.p/hb-style.cc.o.d -o src/libharfbuzz.a.p/hb-style.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-style.cc
[51/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include/libpng16 -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz.a.p/hb-coretext.cc.o -MF src/libharfbuzz.a.p/hb-coretext.cc.o.d -o src/libharfbuzz.a.p/hb-coretext.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-coretext.cc
FAILED: src/libharfbuzz.a.p/hb-coretext.cc.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include/libpng16 -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz.a.p/hb-coretext.cc.o -MF src/libharfbuzz.a.p/hb-coretext.cc.o.d -o src/libharfbuzz.a.p/hb-coretext.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-coretext.cc
../src/3.2.0-8c7a14e743.clean/src/hb-coretext.cc:900:14: error: variable 'status_and' set but not used [-Werror,-Wunused-but-set-variable]
    uint32_t status_and = ~0, status_or = 0;
             ^
1 error generated.
[52/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include/libpng16 -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz.a.p/hb-ot-shape-complex-arabic.cc.o -MF src/libharfbuzz.a.p/hb-ot-shape-complex-arabic.cc.o.d -o src/libharfbuzz.a.p/hb-ot-shape-complex-arabic.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-ot-shape-complex-arabic.cc
[53/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset-input.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset-input.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset-input.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset-input.cc
[54/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-static.cc.o -MF src/libharfbuzz-subset.a.p/hb-static.cc.o.d -o src/libharfbuzz-subset.a.p/hb-static.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-static.cc
[55/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset-cff-common.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset-cff-common.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset-cff-common.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset-cff-common.cc
[56/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-ot-cff1-table.cc.o -MF src/libharfbuzz-subset.a.p/hb-ot-cff1-table.cc.o.d -o src/libharfbuzz-subset.a.p/hb-ot-cff1-table.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-ot-cff1-table.cc
[57/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-ot-cff2-table.cc.o -MF src/libharfbuzz-subset.a.p/hb-ot-cff2-table.cc.o.d -o src/libharfbuzz-subset.a.p/hb-ot-cff2-table.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-ot-cff2-table.cc
[58/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset-cff2.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset-cff2.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset-cff2.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset-cff2.cc
[59/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset-cff1.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset-cff1.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset-cff1.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset-cff1.cc
[60/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/lib/pkgconfig/../../include/libpng16 -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz.a.p/hb-ot-layout.cc.o -MF src/libharfbuzz.a.p/hb-ot-layout.cc.o.d -o src/libharfbuzz.a.p/hb-ot-layout.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-ot-layout.cc
[61/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset-plan.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset-plan.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset-plan.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset-plan.cc
[62/65] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -Isrc/libharfbuzz-subset.a.p -Isrc -I../src/3.2.0-8c7a14e743.clean/src -I. -I../src/3.2.0-8c7a14e743.clean -I/Users/leanderSchulten/git_projekte/Lichtsteuerung/vcpkg_installed/arm64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -fno-rtti -O0 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.2 -O3 -DNDEBUG -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-subset.a.p/hb-subset.cc.o -MF src/libharfbuzz-subset.a.p/hb-subset.cc.o.d -o src/libharfbuzz-subset.a.p/hb-subset.cc.o -c ../src/3.2.0-8c7a14e743.clean/src/hb-subset.cc
ninja: build stopped: subcommand failed.
```
</details>

**Additional context**
<details><summary>vcpkg.json</summary>

```
{
  "name": "lichtsteuerung",
  "version-string": "2.0.0",
  "dependencies": [
    "pango"
  ]
}

```
</details>

--- a/ports/harfbuzz/fix-macos-build.diff
+++ b/ports/harfbuzz/fix-macos-build.diff
@@ -1,0 +1,43 @@
+diff --git a/src/hb-coretext.cc b/src/hb-coretext.cc
+index 5f383064c4..6ccc1b0a2b 100644
+--- a/src/hb-coretext.cc
++++ b/src/hb-coretext.cc
+@@ -897,7 +897,7 @@ _hb_coretext_shape (hb_shape_plan_t    *shape_plan,
+     DEBUG_MSG (CORETEXT, nullptr, "Num runs: %d", num_runs);
+ 
+     buffer->len = 0;
+-    uint32_t status_and = ~0, status_or = 0;
++    uint32_t status_or = 0;
+     CGFloat advances_so_far = 0;
+     /* For right-to-left runs, CoreText returns the glyphs positioned such that
+      * any trailing whitespace is to the left of (0,0).  Adjust coordinate system
+@@ -918,7 +918,6 @@ _hb_coretext_shape (hb_shape_plan_t    *shape_plan,
+       CTRunRef run = static_cast<CTRunRef>(CFArrayGetValueAtIndex (glyph_runs, i));
+       CTRunStatus run_status = CTRunGetStatus (run);
+       status_or  |= run_status;
+-      status_and &= run_status;
+       DEBUG_MSG (CORETEXT, run, "CTRunStatus: %x", run_status);
+       CGFloat run_advance = CTRunGetTypographicBounds (run, range_all, nullptr, nullptr, nullptr);
+       if (HB_DIRECTION_IS_VERTICAL (buffer->props.direction))
+@@ -1140,21 +1139,6 @@ _hb_coretext_shape (hb_shape_plan_t    *shape_plan,
+       buffer->len += num_glyphs;
+     }
+ 
+-    /* Mac OS 10.6 doesn't have kCTTypesetterOptionForcedEmbeddingLevel,
+-     * or if it does, it doesn't respect it.  So we get runs with wrong
+-     * directions.  As such, disable the assert...  It wouldn't crash, but
+-     * cursoring will be off...
+-     *
+-     * https://crbug.com/419769
+-     */
+-    if (false)
+-    {
+-      /* Make sure all runs had the expected direction. */
+-      HB_UNUSED bool backward = HB_DIRECTION_IS_BACKWARD (buffer->props.direction);
+-      assert (bool (status_and & kCTRunStatusRightToLeft) == backward);
+-      assert (bool (status_or  & kCTRunStatusRightToLeft) == backward);
+-    }
+-
+     buffer->clear_positions ();
+ 
+     unsigned int count = buffer->len;

--- a/ports/harfbuzz/portfile.cmake
+++ b/ports/harfbuzz/portfile.cmake
@@ -11,6 +11,7 @@ vcpkg_from_github(
         # Details: https://github.com/microsoft/vcpkg/issues/16262
         0001-circumvent-samefile-error.patch
         0002-fix-uwp-build.patch
+        fix-macos-build.diff # fixes https://github.com/harfbuzz/harfbuzz/issues/3484
 )
 
 if("icu" IN_LIST FEATURES)

--- a/ports/harfbuzz/vcpkg.json
+++ b/ports/harfbuzz/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "harfbuzz",
   "version-semver": "3.2.0",
-  "port-version": 1,
+  "port-version": 2,
   "description": "HarfBuzz OpenType text shaping engine",
   "homepage": "https://github.com/harfbuzz/harfbuzz",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2710,7 +2710,7 @@
     },
     "harfbuzz": {
       "baseline": "3.2.0",
-      "port-version": 1
+      "port-version": 2
     },
     "hayai": {
       "baseline": "2019-08-10",

--- a/versions/h-/harfbuzz.json
+++ b/versions/h-/harfbuzz.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "fb0fcb23cc704f5a2bca6fc631622309876ab271",
+      "version-semver": "3.2.0",
+      "port-version": 2
+    },
+    {
       "git-tree": "b2c918c04e1ecf5b6a591911cea03f7093ab6933",
       "version-semver": "3.2.0",
       "port-version": 1

